test/name_service/utils.clj

```diff
@@ -1,2 +1,26 @@
 (ns name-service.utils
-  (:require [clojure.test :refer :all]))
+  (:require [clojure.test :refer :all]
+            [name-service.core.entity :as entity]))
+
+(defn- fetch
+  [db key val]
+  (first (filter #(= val (get % key)) @db)))
+
+(defn- delete
+  [db key val]
+  (let [keymap (fetch db key val)]
+    (swap! db disj keymap)))
+
+(defn- save
+  [db keymap key val]
+  (let [old-keymap (fetch db key val)]
+    (if (> (count old-keymap) 0)
+      (delete db key val)))
+  (swap! db conj keymap)
+  keymap)
+
+(defrecord TestStorage [db]
```

Comment:
This storage expects (and works only with) atoms with value of a set. Document this.